src/pages/iou/steps/NewRequestAmountPage.js

```diff
@@ -0,0 +1,242 @@
+        const activeRoute = encodeURIComponent(Navigation.getActiveRoute().replace(/\?.*/, ''));
+        Navigation.navigate(ROUTES.getMoneyRequestCurrencyRoute(iouType, reportID, currency, activeRoute));
+    };
+
+    const navigateToNextPage = (currentAmount) => {
+        const amountInSmallestCurrencyUnits = CurrencyUtils.convertToSmallestUnit(currency, Number.parseFloat(currentAmount));
+        IOU.setMoneyRequestAmount(amountInSmallestCurrencyUnits);
+        IOU.setMoneyRequestCurrency(currency);
+
+        if (isEditing) {
+            Navigation.goBack(ROUTES.getMoneyRequestConfirmationRoute(iouType, reportID));
+            return;
+        }
+
+        const moneyRequestID = `${iouType}${reportID}`;
+        const shouldReset = iou.id !== moneyRequestID;
+        // If the money request ID in Onyx does not match the ID from params, we want to start a new request
+        // with the ID from params. We need to clear the participants in case the new request is initiated from FAB.
+        if (shouldReset) {
+            IOU.setMoneyRequestId(moneyRequestID);
+            IOU.setMoneyRequestDescription('');
+            IOU.setMoneyRequestParticipants([]);
+        }
+
+        // If a request is initiated on a report, skip the participants selection step and navigate to the confirmation page.
+        if (report.reportID) {
+            // Reinitialize the participants when the money request ID in Onyx does not match the ID from params
+            if (_.isEmpty(participants)) {
+                const currentUserAccountID = currentUserPersonalDetails.accountID;
+                const iouParticipants = ReportUtils.isPolicyExpenseChat(report)
+                    ? [{reportID: report.reportID, isPolicyExpenseChat: true, selected: true}]
+                    : _.chain(report.participantAccountIDs)
+                          .filter((accountID) => currentUserAccountID !== accountID)
+                          .map((accountID) => ({accountID, selected: true}))
+                          .value();
+                IOU.setMoneyRequestParticipants(iouParticipants);
+            }
+            Navigation.navigate(ROUTES.getMoneyRequestConfirmationRoute(iouType, reportID));
+            return;
+        }
+        Navigation.navigate(ROUTES.getMoneyRequestParticipantsRoute(iouType));
+    };
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            onEntryTransitionEnd={focusTextInput}
+        >
+            {({safeAreaPaddingBottomStyle}) => (
+                <FullPageNotFoundView shouldShow={!IOUUtils.isValidMoneyRequestType(iouType)}>
+                    <View style={[styles.flex1, safeAreaPaddingBottomStyle]}>
+                        {isEditing && (
+                            <HeaderWithBackButton
+                                title={titleForStep}
+                                onBackButonBackButtonPress={navigateBack}
+                            />
+                        )}
```

Comment:
I didn't review in depth but I'm pretty sure we have a wrong merge conflicts here. We merged this few hours ago https://github.com/Expensify/App/pull/24097

Reply:
ok, thanks @s77rt! I'll fix this